floyd/cli/data.py

```diff
@@ -159,18 +161,30 @@ def delete(ids, yes):
             failures = True
             continue
 
-        if not yes and not click.confirm("Delete Data: {}?".format(data_source.name),
+        data_name = normalize_data_name(data_source.name)
+        suffix = data_name.split('/')[-1]
+        if not suffix.isdigit():
+            failures = True
+            floyd_logger.error('%s is not a dataset, skipped.', id)
```

Comment:
If the name is not ending with version number, we can already conclude that it's not an dataset. I added it here under the assumption that we will have more variants other than `output` in the future.